### PR TITLE
feat/800/poetry-dependency-management: introduce poetry

### PR DESCRIPTION
**Pull request by jkrzemin**
Moves dependency and project management from setup.py to poetry.

Fixes issue #800 

## Proposed Changes

- Introduce kapitan to poetry using PEP-517
- Remove requirements
- Remove setup.py
- Clean up dependencies

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,70 @@
...
+    "Development Status :: 4 - Beta",
+    "Intended Audience :: Developers",
+    "Topic :: Software Development :: Build Tools",
+    "License :: OSI Approved :: Apache Software License",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+]
+license = "Apache-2.0"
+version = "0.31.0rc0"
```
Comment (ramaro) [inline pyproject.toml:28]: Just one fix: version should now be `0.31.0`

Comment (jkrzemin) [reply pyproject.toml]: Alright, fixed. I would also like to use something like https://github.com/tiangolo/poetry-version-plugin in next step.

Comment (ramaro) [reply pyproject.toml]: Nice one, didn't know about it! That's a great idea :D

**Comment by ramaro** [comment]
@jkrzemin this looks good thank you! Unfortunately we've done some requirements.txt updates along the way - I can submit those updates to your branch to fix the conflicts if you want 👍🏼

File: pyproject.toml (hunk 1 above)
Comment (jkrzemin) [reply pyproject.toml]: I've used similar solution in my personal projects, but there I've used https://flit.pypa.io/en/stable/

https://smarie.github.io/python-getversion/

**Comment by jkrzemin** [comment]
> 
I'll update it by tomorrow

**Comment by jkrzemin** [comment]
@ramaro  alright, I've updated the requirements in pyproject.toml - I've checked all packages added in requirements.txt within this year.

**Comment by ramaro** [comment]
Perfect, thank you @jkrzemin 👍🏼

**Comment by Moep90** [comment]
Hi @jkrzemin,

I noticed that your pull request was already merged, but unfortunately, it lacked documentation, and an outdated Dockerfile which has resulted in a failing build pipeline. In the future, it would be helpful if you could prioritize documentation to ensure that our code is easily maintainable, and our build pipeline is successful.

Moving forward, I would recommend that you update the code with the necessary documentation as soon as possible to ensure that our project code is up to our standards #941 . @ramaro It may also be worth considering adding another step to our review process to ensure that all changes include proper documentation before merging.

Thank you for your contributions, and I appreciate your efforts and improvements.

cc: @MatteoVoges

**Pull request merged**